superset-frontend/src/SqlLab/components/RunQueryActionButton.tsx

```diff
@@ -17,53 +17,55 @@
  * under the License.
  */
 import React from 'react';
-import PropTypes from 'prop-types';
 import { t } from '@superset-ui/translation';
 
 import Button from '../../components/Button';
 
-const propTypes = {
-  allowAsync: PropTypes.bool.isRequired,
-  dbId: PropTypes.number,
-  queryState: PropTypes.string,
-  runQuery: PropTypes.func.isRequired,
-  selectedText: PropTypes.string,
-  stopQuery: PropTypes.func.isRequired,
-  sql: PropTypes.string.isRequired,
-};
-const defaultProps = {
-  allowAsync: false,
-  sql: '',
-};
+const NO_OP = () => undefined;
+
+interface Props {
+  allowAsync: boolean;
+  dbId?: number;
+  queryState: string;
+  runQuery: (c?: boolean) => void;
+  selectedText?: string;
+  stopQuery: () => void;
+  sql: string;
+}
 
-export default function RunQueryActionButton(props) {
-  const runBtnText = props.selectedText
-    ? t('Run Selected Query')
-    : t('Run Query');
-  const btnStyle = props.selectedText ? 'warning' : 'primary';
-  const shouldShowStopBtn =
-    ['running', 'pending'].indexOf(props.queryState) > -1;
+const RunQueryActionButton = ({
+  allowAsync = false,
+  dbId,
+  queryState = '',
+  runQuery = NO_OP,
+  selectedText,
+  stopQuery = NO_OP,
+  sql,
+}: Props) => {
+  const runBtnText = selectedText ? t('Run Selected Query') : t('Run Query');
+  const btnStyle = selectedText ? 'warning' : 'primary';
+  const shouldShowStopBtn = ['running', 'pending'].indexOf(queryState) > -1;
```

Comment:
you can change this to `const shouldShowStopBtn = !!queryState && ['running', 'pending'].indexOf(queryState) > -1
`

Then you won't need to default queryState to empty string

Reply:
Done.